app/views/articles/_search.html.erb

```diff
@@ -45,68 +43,64 @@
         var className = e.target.className;
         for (var i = 0; i < filterButts.length; i++) {
           filterButts[i].classList.remove("selected");
         }
         if (className.indexOf("selected") == -1) {
           e.target.classList.add("selected");
-          window.history.replaceState(null, null, "/search?q=" + query + "&filters=" + filters);
-          search(query, index, filters);
+          window.history.replaceState(null, null, "/search?search_fields=" + query + "&filters=" + filters);
+          search(query, filters);
         } else {
-          window.history.replaceState(null, null, "/search?q=" + query);
-          search(query, index, "");
+          window.history.replaceState(null, null, "/search?search_fields=" + query);
+          search(query, "");
         }
       }
     }
   }
 
-  function search(query, index, filters) {
-    var hashtags = query.match(/#\w+/g)
-    var searchObj = {
-      hitsPerPage: 60,
-      page: "0",
-      queryType: "prefixNone",
-      attributesToRetrieve: [
-        'id',
-        'title',
-        'path',
-        'class_name',
-        'comments_count',
-        'tag_list',
-        'readable_publish_date',
-        'positive_reactions_count',
-        'flare_tag',
-        'user',
-        'reading_time'
-      ],
-      attributesToHighlight: [],
-      exactOnSingleWordQuery: "none",
-      attributesToSnippet: ['body_text:19', 'comments_blob:8'],
-    }
+  function search(query, filters) {
+    var hashtags = query.match(/#\w+/g);
+    var searchTerm = query.replace(/#/g, '').trim();
+    var searchHash = { per_page: 60, page: 0 };
 
-    if (filters === "MY_POSTS" && userData()) {
-      searchObj["tagFilters"] = [['user_' + userData()['id']]]
-    } else if (hashtags != null && hashtags.length > 0) {
+    if (hashtags && hashtags.length > 0) {
       for(var i = 0; i < hashtags.length; i++){
-        hashtags[i] = hashtags[i].replace(/#/,'')
+        hashtags[i] = hashtags[i].replace(/#/,'');
       }
-      searchObj["tagFilters"] = hashtags
-    } else {
-      searchObj["filters"] = filters
+      searchHash["tag_names"] = hashtags;
+    } else if (filters && filters !== "") {
+      filters.split('&').forEach((filter) => {
+        const [key, value] = filter.split(':');
+        searchHash[key] = value;
+      });
     }
-    index.search(query, searchObj)
-      .then(function searchDone(content) {
+
+    if (searchTerm && searchTerm !== "") { searchHash["search_fields"] = searchTerm; }
+
+    const searchParams = new URLSearchParams(searchHash).toString();
+
+    fetch(`/search/feed_content?${searchParams}`, {
+      method: 'GET',
+      headers: {
+        Accept: 'application/json',
+        'X-CSRF-Token': window.csrfToken,
+        'Content-Type': 'application/json',
+      },
+      credentials: 'same-origin',
+    })
+      .then(response => response.json())
```

Comment:
Can we use the [`fetchSearch` helper function](https://github.com/atsmith813/dev.to/blob/9a0a6fbe8f3b45df600ba3a1571ab8164fd1bc41/app/javascript/src/utils/search.js#L90-L102) here?

```javascript
const responsePromise = fetchSearch('feed_content', searchHash)

responsePromise.then((content) => {
  // continuing on...
}
```

Reply:
You could even dare to try `async/await` as it's now been enabled:

```js
const result = await fetchSearch(...);
// do something with the result
```

I'll double check to see if it actually works :D

Reply:
@atsmith813 we cant use that helper here or in the other file for the following reasons as explained to me by @rhymes 

> I don’t think it’ll work but better check first. The reason why it doesn’t is that the code you’re trying to refactor by using fetchSearch  is handled by sprockets, but the fetchSearch resides in our webpack code, which is imported only at will and not known by app/assets/javascript . Long story short. app/javascript  can import code from app/assets/javascripts  but not the other way around
the only thing you can do is to move the function inside app/assets/javascripts  and call it from both places but i think it’s outside of the scope of the PR

Reply:
Makes sense. That's kind of funny, that's why I didn't include the import statement in the code example and just wrote _"after importing it"_ because I realized I was unclear on how that would actually be imported 🤔TIL and this is a good example of [our worst technical debt](https://docs.dev.to/technical-overview/architecture/#worst-technical-debt). Thanks for the info, both of you!